cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -29,14 +32,37 @@ private[lagom] object JoinClusterImpl {
     if (cluster.settings.SeedNodes.isEmpty && joinSelf)
       cluster.join(cluster.selfAddress)
 
+    val SUCCESSFUL_EXIT = 0
+    val exitStatus = new AtomicInteger(SUCCESSFUL_EXIT)
+
+    val isExiting = new AtomicBoolean(false)
+
     if (exitJvm) {
       // exit JVM when ActorSystem has been terminated
-      system.registerOnTermination(System.exit(0))
+      system.registerOnTermination {
+        new Thread("Sys-exiting-from-akka-termination") {
```

Comment:
Another idea is that we spawn a new thread here, but in that thread we `Await.ready(system.whenTerminated, timeout)` before calling `System.exit`. The `whenTerminated` should be completed later.

Reply:
👍

Reply:
So was there a decision _not_ to fix this?

Reply:
@TimMoore we decided we wouldn't invest more time on improving and testing this class before release `1.3.8` and would postpone further improvements to a separate PR. (see @patriknw comment above where he suggests we stop efforts here as it'll be obsolete when moving to Akka 2.5).

Despite that, I started https://github.com/lagom/lagom/pull/978 to share some ideas I had while working on this improvement.

Reply:
well, creating `new Thread` without starting it looks like a bug to me, if you don't need to start a new thread then you should use the old syntax:
```
system.registerOnTermination {
  if (isExiting.compareAndSet(false, true)) {
  ...
}
```

Reply:
@patriknw I think this was not supposed to be merged. I spent some time with @ignasi35 yesterday discussing the details and we came to the conclusion that we needed a new PR to improve it. I guess our intention was to call `.start()` but we end up merging it. 

Although the code is now misleading, it does produces the desired effect. Maybe in a less efficient fashion, but it is not malfunctioning, AFAICT.

This will be revised soon also for 1.3.x series.